sites/partners/lib/formatApplicationData.ts

```diff
@@ -184,10 +190,15 @@ export const mapFormToApi = ({ data, listingId, editMode, programs }: mapFormToA
 
   // we need to add primary applicant
   const householdSize = householdMembers.length + 1 || 1
+  let preferredUnit: preferredUnitType[] = []
```

Comment:
@YazeedLoonat , rather than defined `preferredUnitType` above, we should be able to use `Record<"id", string>[]` here. Can you make that change and then I think it's good to merge.